src/python/WMCore/Storage/Backends/GFAL2Impl.py

```diff
@@ -24,10 +26,43 @@ def __init__(self, stagein=False):
         # Next commands after separation are executed without env -i and this leads us with
         # mixed environment with COMP and system python.
         # GFAL2 is not build under COMP environment and it had failures with mixed environment.
-        self.setups = "env -i X509_USER_PROXY=$X509_USER_PROXY JOBSTARTDIR=$JOBSTARTDIR bash -c '{}'"
+        self.setups = "env -i JOBSTARTDIR=$JOBSTARTDIR bash -c '{}'"  # Default initialization, tweaked afterwards
         self.removeCommand = self.setups.format('. $JOBSTARTDIR/startup_environment.sh; date; gfal-rm -t 600 {}')
         self.copyOpts = '-t 2400 -T 2400 -p -v --abort-on-failure {checksum} {options} {source} {destination}'
-        self.copyCommand = self.setups.format('. $JOBSTARTDIR/startup_environment.sh; date; gfal-copy ' + self.copyOpts)
+        self.copyCommand = self.setups.format('. $JOBSTARTDIR/startup_environment.sh; date; gfal-copy -vvv ' + self.copyOpts)
+
+    def adjustSetup(self, authmethod=None, forcemethod=False):
+        """
+        Adjust the `self.setups` based on the selected authentication method and regenerate commands.
+        :authmethod: str, the authentication method to be preferentially used ("X509", "TOKEN", or None)
+        :forcemethod: bool, whether to force the use of the preferred authentication method, disabling the other
+        """
+        if authmethod == "X509":
```

Comment:
I feel like it will be easy to get confused and/or make mistakes on this code, caused by the variable overrides and duplication. 
To make it more sustainable, I would suggest not to override/redefine anything that is not explicitly needed, for instance `self.copyOpts` could be skipped, AFAICT. But there might be other things that we could skip in this method as well.